### ISSUE TITLE
Rule proposal (Solid 2.0): `await` inside an `action()` generator must be followed by a bare `yield`

### ISSUE THREAD

**Issue by ryansolid** [solidjs-community/eslint-plugin-solid#229]
## Summary

Solid 2.0's `action()` wraps an (async) generator so each `yield` is a transaction step: the runtime re-enters the action's transaction before running the code after a `yield`. A plain `await` is invisible to the runtime — it has no hook into an async generator's internal `await` continuations — so **code between an `await` and the next `yield` runs outside the transaction.** Writes there commit immediately instead of with the action, and anything that *creates a reader* there (`until()`, `latest()`, `createMemo`/effects, a component mount) is created mainline; reading the action's held state, such a reader is "born held" and replays only at the action's commit. For `until()` that commit is the settle its own promise holds open — the action deadlocks (solidjs/solid#3482).

The documented idiom is a bare `yield` right after the `await`:

```ts
const saved = await api.createTodo(text); // typed result
yield; // re-enter the transaction
setTodos(t => { ... });
yield until(() => todos.some(t => t.id === saved.id));
```

The runtime cannot detect the omission, and the failure mode is a hang or a `TimeoutError` far from the cause. This is a good fit for a lint rule.

## Proposed rule: `solid/action-yield-after-await`

Inside the generator function passed to `action(...)` (solid-js export; also the `createForm`/`createAction`-style wrappers that build on it, if the plugin tracks those):

> Every `await` expression whose continuation belongs to the generator itself must be **immediately followed by a bare `yield;` statement**, or be the operand of a `yield` (`yield somePromise`).

Deliberately conservative — decidable syntactically, no false negatives:

- Flags `await x; yield until(...)` — the `until(...)` operand is evaluated *before* the second `yield` re-enters the transaction, so it is not a valid step boundary for the code that precedes it.
- Flags `const r = await x; setFoo(r)` (write outside the transaction).
- Allows `const r = yield api.call()` (untyped, but a step).
- Allows `await x; yield; ...`.
- `await` inside a nested function/arrow/callback within the body is **not** the generator's continuation and is ignored.
- Control flow is fine because the rule is local to each `await` statement (each branch's `await` needs its own following `yield;`).

**Autofix:** insert `yield;` as the next statement after the `await`-containing statement. For `const x = await …` the fix goes after the declaration. A bare `yield` where nothing reactive follows is harmless (an empty step), which is what makes the conservative rule safe to autofix.

**Message:** "`await` inside an `action()` runs the code after it outside the transaction; add a bare `yield;` before any write or reader creation (Solid 2.0 `action` docs)."

## Scope notes

- Only generator functions that are the direct argument of an `action(...)` call imported from `solid-js` (or a configured list of wrapper names). A generator assigned to a variable and passed later is a reasonable follow-up.
- Sync generators (`function*`) can't `await`; the rule only has work to do on `async function*`.
- The precise rule ("no write or reader creation between `await` and `yield`") is *not* safely lintable — whether `doThing()` creates a reader isn't knowable statically — which is why the proposal is the strict form.

Happy to review a PR or answer questions about the runtime side. Reference: [`action` docstring](https://github.com/solidjs/solid/blob/next/packages/signals/src/core/action.ts), solidjs/solid#3482, solidjs/solid#3491 (docs fix).
